verilog/analysis/checkers/parameter_name_style_rule.cc

```diff
@@ -92,43 +105,43 @@ void ParameterNameStyleRule::HandleSymbol(const verible::Symbol &symbol,
   if (ParamDeclMatcher().Matches(symbol, &manager)) {
     if (IsParamTypeDeclaration(symbol)) return;
 
-    const auto param_decl_token = GetParamKeyword(symbol);
+    const verilog_tokentype param_decl_token = GetParamKeyword(symbol);
 
     auto identifiers = GetAllParameterNameTokens(symbol);
 
     for (const auto *id : identifiers) {
-      const auto param_name = id->text();
-      uint32_t observed_style = 0;
-      if (verible::IsUpperCamelCaseWithDigits(param_name)) {
-        observed_style |= kUpperCamelCase;
-      }
-      if (verible::IsNameAllCapsUnderscoresDigits(param_name)) {
-        observed_style |= kAllCaps;
-      }
-      if (param_decl_token == TK_localparam && localparam_allowed_style_ &&
-          (observed_style & localparam_allowed_style_) == 0) {
-        violations_.insert(LintViolation(
-            *id, ViolationMsg("localparam", localparam_allowed_style_),
-            context));
-      } else if (param_decl_token == TK_parameter && parameter_allowed_style_ &&
-                 (observed_style & parameter_allowed_style_) == 0) {
-        violations_.insert(LintViolation(
-            *id, ViolationMsg("parameter", parameter_allowed_style_), context));
+      const auto name = id->text();
+      switch (param_decl_token) {
+        case TK_localparam:
+          if (!RE2::FullMatch(name, *localparam_style_regex_)) {
+            violations_.insert(LintViolation(
+                *id, CreateLocalparamViolationMessage(), context));
+          }
+          break;
+
+        case TK_parameter:
+          if (!RE2::FullMatch(name, *parameter_style_regex_)) {
+            violations_.insert(
+                LintViolation(*id, CreateParameterViolationMessage(), context));
+          }
+          break;
+
+        default:
+          break;
       }
     }
   }
 }
 
 absl::Status ParameterNameStyleRule::Configure(
     absl::string_view configuration) {
-  // TODO(issue #133) include bitmap choices in generated documentation.
-  static const std::vector<absl::string_view> choices = {
-      "CamelCase", "ALL_CAPS"};  // same sequence as enum StyleChoicesBits
-  using verible::config::SetNamedBits;
-  return verible::ParseNameValues(
+  using verible::config::SetRegex;
+  absl::Status s = verible::ParseNameValues(
       configuration,
-      {{"localparam_style", SetNamedBits(&localparam_allowed_style_, choices)},
-       {"parameter_style", SetNamedBits(&parameter_allowed_style_, choices)}});
+      {{"localparam_style_regex", SetRegex(&localparam_style_regex_)},
```

Comment:
We shouldn't remove the old configuration names: if someone has used localparam_style and parameter_style in their configuration it will suddenly fail as it doesn't exist anymore.

Instead, we should have not only the new regex names, but also the old names, which then select a regex.
So someone can set `localparam_style_regex`, but they can also say  `localparam_style=ALL_CAPS`, which could be considered a name for a regex; So if such value has been set, we then then can set the upper case regex.

Reply:
Do you have a preference on how it should handle if both `localparam_style` and `localparam_style_regex` are set with valid values? 

Maybe have `ParameterNameStyleRule::Configure` return an `absl::InvalidArgumentError()` noting the user should use one or the other.

Reply:
Ah, yes, the OR-ing part, I didn't think of it: yes that should then assemble a `(foo)|(bar)` kindof regex.

But with OR-ing in place, we can assemble all the regex from `localparam_style` with  the `localparam_style_regex` if both of them are given in the lint configuration.

Probably good to have a test for all these situations; for that, you could have public functions that return the `const RE2 *` for each regex
```c++
const RE2 *localparam_style_regex() const { return localparam_style_regex_.get(); }
```
, so that it is easy to write a test that verifies that the returned value is `re != nullptr` and the `re->pattern()` matches the expectation after having called `Configuration()`.